pycloudlib/gce/cloud.py

```diff
@@ -50,6 +52,21 @@ def __init__(
 
         if project:
             os.environ["GOOGLE_CLOUD_PROJECT"] = str(project)
+        else:
```

Comment:
@lucasmoura Still not exactly sure how this env var is being used, so not sure if I should be setting it if we obtain the project id from the gcloud tool.

Reply:
@TheRealFalcon the GOOGLE_CLOUD_PROJECT env  is used if gcloud config wasn't set up on the jenkins worker node, so I think we still need both paths as some clean instances would not have those credentials setup locally.

Reply:
I think @TheRealFalcon has a point here. Looking at the docs, we look first for the `project-id` on the `GOOGLE_CLOUD_PROJECT` variable. If we do not find it, then we search for the `GOOGLE_CLOUD_CREDENTIALS`

https://googleapis.dev/python/google-cloud-core/latest/config.html

Because we must setup the `GOOGLE_CLOUD_CREDENTIALS`, I think we can skip setting this env variable